dockstore-webservice/src/main/java/io/dockstore/webservice/Helper.java

```diff
@@ -442,6 +547,147 @@ public String getContent() {
         return new ArrayList(userDAO.findById(userId).getContainers());
     }
 
+    public static FileResponse readGitRepositoryFile(Container container, String fileName, HttpClient client) {
+        FileResponse cwl = new Helper.FileResponse();
+
+        Pattern p = Pattern.compile("git\\@(\\S+):(\\S+)/(\\S+)\\.git");
+        Matcher m = p.matcher(container.getGitUrl());
+        if (!m.find()) {
+            throw new WebApplicationException(HttpStatus.SC_NOT_FOUND);
+        }
+
+        // These correspond to the positions of the pattern matcher
+        final int sourceIndex = 1;
+        final int usernameIndex = 2;
+        final int reponameIndex = 3;
+
+        String source = m.group(sourceIndex);
+        String gitUsername = m.group(usernameIndex);
+        String gitRepository = m.group(reponameIndex);
+        LOG.info("Source: " + source);
+        LOG.info("Username: " + gitUsername);
+        LOG.info("Repository: " + gitRepository);
+
+        // TODO: this only works with public repos, we will need an endpoint for public and another for auth to handle private repos in the
+        // future
+        // search for the Dockstore.cwl
+        if (source.equals("github.com")) {
```

Comment:
These two branches should be refactored into a new interface with two implementations (bitbucket and github), this will make this code more readable and give future developers a better idea of what is required to implement a new source control system.